### PR TITLE
fix: match mono-service gas throttling

### LINKED ISSUE #11854: Expected but did not find [transactionFee, contractCallResult] (round 162907351)
### Description

Mismatched field counts  (expected but not find [transactionFee, contractCallResult])

Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708327150.295352242)

Transaction body:
```
transactionID {
  transactionValidStart {
    seconds: 1708327141
    nanos: 329996922
  }
  accountID {
    accountNum: 3064793
  }
}
nodeAccountID {
  accountNum: 17
}
transactionFee: 500000000
transactionValidDuration {
  seconds: 120
}
memo: "ca2c910b12bbdf47babded581e14bdccd32c7b204a"
contractCall {
  contractID {
    contractNum: 3158574
  }
  gas: 450000
  amount: 11462870826
  functionParameters: "\202XY\331\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\002\255G2,\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000`\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000e\323\001\v\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\005\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\026;Z\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000B\'\027\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\v*\325\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000B\235\350\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\026;Z"
}
```

### Steps to reproduce

Re-run events from round 162907351

### Hedera network

other

### Version

11805-diff-testing branch (with reverted commit 6954892)

### Operating system

macOS

### LINKED ISSUE #11870: TokenAssociateToAccount ≠ ContractCreate DIFFERED (TRANSACTION_MISMATCH)
### Description

Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708318990.065090003)

### Steps to reproduce

Re-run events from round 162898990

### Additional context

_No response_

### Hedera network

other

### Version

11852-pay-rewards-in-triggered-txn

### Operating system

macOS

### LINKED ISSUE #11871: TokenAssociateToAccount DIFFERED (FIRST_IS_MISSING)
### Description

Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708319723.431595003)

Transaction:
```
transactionID {
  transactionValidStart {
    seconds: 1708319710
    nanos: 760553933
  }
  accountID {
    accountNum: 4340526
  }
}
nodeAccountID {
  accountNum: 4
}
transactionFee: 500000000
transactionValidDuration {
  seconds: 120
}
tokenAssociate {
  account {
    accountNum: 4340526
  }
  tokens {
    tokenNum: 127877
  }
}

```

### Steps to reproduce

Re-run events from round 162898990

### Additional context

_No response_

### Hedera network

other

### Version

11852-pay-rewards-in-triggered-txn (commit 67210a)

### Operating system

macOS

### LINKED ISSUE #11890: Entries at position 974740 had different transactions (signedTransactionBytes)
### Description

Error message:
```
Entries at position 974740 had different transactions (signedTransactionBytes: "\n%\n\026\n\f\b\300\265\315\256\006\020\200\372\334\355\001\022\004\030\200\342< \001B\vB\005\b\200\316\332\003z\002\030\000"
 vs signedTransactionBytes: "\n!\n\026\n\f\b\300\265\315\256\006\020\200\372\334\355\001\022\004\030\200\342< \001B\aB\005\b\200\316\332\003")
```

Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708350157.001612003)

Transaction Body:
```
For body,
transactionID {
  transactionValidStart {
    seconds: 1708350144
    nanos: 498547968
  }
  accountID {
    accountNum: 995584
  }
  nonce: 1
}
contractCreateInstance {
  autoRenewPeriod {
    seconds: 7776000
  }
  auto_renew_account_id {
    accountNum: 0
  }
}
```

### Steps to reproduce

Re-run events from round 162933431

### Additional context

_No response_

### Hedera network

other

### Version

11852-pay-rewards-in-triggered-txn

### Operating system

macOS

### LINKED ISSUE #11912: (02/19 - 14:00:00) expected: <INSUFFICIENT_GAS> but was: <REVERTED_SUCCESS>
Transaction in HashScan [here](https://hashscan.io/mainnet/transaction/1708351275.035156635).

A contract transaction should have failed with INSUFFICIENT_GAS but didn't.

Mismatched values, expected 'INSUFFICIENT_GAS', got 'REVERTED_SUCCESS' - Matching field 'status' Matching field 'receipt'  ==> expected: <INSUFFICIENT_GAS> but was: <REVERTED_SUCCESS>

For body,
```
transactionID {
  transactionValidStart {
    seconds: 1708351262
    nanos: 799545355
  }
  accountID {
    accountNum: 1109176
  }
  nonce: 12
}
contractCall {
  contractID {
    contractNum: 359
  }
  gas: 1
  functionParameters: "a\215\306^\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\006\370\232p\240\2021\000\000\000\000\000\000\000\000\000\000\000\0006\254\337\341\313\371\t\213\333z<b\270\352\241\001l\021\0361"
}
```
This was run against branch 11852-pay-rewards-in-triggered-txn on 3/5

### LINKED ISSUE #11913: (2/29 14:15) Expected but not find [transactionFee, contractCallResult] (round 162936556)
Hashscan tx: [here](https://hashscan.io/mainnet/transaction/1708353171.666850003)

Transaction body:
```
transactionID {
  transactionValidStart {
    seconds: 1708353158
    nanos: 271281965
  }
  accountID {
    accountNum: 3064793
  }
}
nodeAccountID {
  accountNum: 4
}
transactionFee: 500000000
transactionValidDuration {
  seconds: 120
}
memo: "ca0732b8b2bfb040bca55c3e64e050c169622600ag"
contractCall {
  contractID {
    contractNum: 3158574
  }
  gas: 400000
  amount: 865610859568
  functionParameters: "\202XY\331\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\311\214\016W\201\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000`\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000e\323f\261\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\004\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\026;Z\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\v*\325\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\006\370\232\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\026;Z"
}
```

Appears to be the same issue as #11905 and #11879

This was run against branch 11852-pay-rewards-in-triggered-txn on 3/5

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #11854 
 - Closes #11870 
 - Closes #11871 
 - Closes #11890
 - Closes #11912 
 - Closes #11913  
 - Migrate initial throttle snapshots from mono-service state.
     * But leave congestion level starts alone until diff testing proves a need to migrate them too.
 - Remove the slightly misleading `payer` argument from `trackFeePayments()`; this method just tracks the work of charging due diligence or insufficient transaction fee payments, and doesn't need the payer id.
 - Slightly re-order `HandleWorkflow` steps to simulate mono-service fee calculation and throttling:
     1. First validate signatures (calling `trackFeePayments()` on any failure).
     2. Only then track throttle utilization based on the top-level transaction (which also ensures the congestion multiplier will match mono-service, if applicable).
 - Add a `ComputeDispatchFeesAsTopLevel` option to include `memo` size in `bpt` usage for scheduled dispatches only.
 - ‼️  Revert only children of the `childRecordBuilder` after a failed dispatch, **not** all children of the `Savepoint` that performed the dispatch.
 - Revert records in `WorldUpdater.revert()` instead of in `MessageProcessor` implementations.
 - Only include an `autoRenewAccountId` in a synthetic `ContractCreateTransactionBody` if not `0.0.0`.
 - Fix mono-service not to set output bytes on an `INSUFFICIENT_GAS` record, since strictly speaking no call executed in this case.
 - Clean up externalization of `INSUFFICIENT_GAS` records in `HtsSystemContract`; especially for a view call.
 - Temporarily reject aliased `AccountID`'s in `CryptoApproveAllowance`, purely for diff testing.

### CODE REVIEW

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM ! Thanks @tinker-michaelj

**Review by iwsimon — APPROVED** [review]
LGTM. Thanks @tinker-michaelj

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by iwsimon — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/NetworkUtilizationManager.java [hunk 1]
```diff
@@ -50,8 +49,7 @@ void trackTxn(
      * @param consensusNow - the consensus time of the transaction.
      * @param state - the state of the node.
      */
-    void trackFeePayments(
-            @NonNull AccountID payer, @NonNull final Instant consensusNow, @NonNull final HederaState state);
+    void trackFeePayments(@NonNull final Instant consensusNow, @NonNull final HederaState state);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/NetworkUtilizationManager.java:52]: At L48 remove parameter javadoc `payer`

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/ChildFeeContextImpl.java [hunk 2]
```diff
@@ -43,16 +43,19 @@ public class ChildFeeContextImpl implements FeeContext {
...
     private final AccountID payerId;
+    private final boolean computeFeesAsInternalDispatch;
 
     public ChildFeeContextImpl(
             @NonNull final FeeManager feeManager,
             @NonNull final HandleContextImpl context,
             @NonNull final TransactionBody body,
-            @NonNull final AccountID payerId) {
+            @NonNull final AccountID payerId,
+            final boolean computeFeesAsInternalDispatch) {
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/ChildFeeContextImpl.java:53]: I know this is just temporary (for diff testing) but it still would have been nice if this variable was the same "direction" as `ComputeDispatchFeesAsTopLevel`, i.e., `ComputeDispatchFeesAsTopLevel.YES` == `true` here.  Instead, it's the reverse.  Unfortunate mental gymnastics.

(You see the direction change over at `HandleContextInfo` L529.)

**Review by david-bakin-sl — COMMENTED** [review]
Quick unfinished look, now waiting for our scheduled call this afternoon.

**Review by lukelee-sl — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/NetworkUtilizationManager.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/NetworkUtilizationManager.java:52]: Will do.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/ChildFeeContextImpl.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/ChildFeeContextImpl.java:53]: Agreeed.

**Pull request merged**